modules/cli/src/test/scala/cli/tests/SetupScalaCLITests.scala

```diff
@@ -32,4 +32,36 @@ class SetupScalaCLITests extends munit.FunSuite {
       System.setProperties(currentProps)
     )
   }
+
+  test(s"should read java properties from JAVA_OPTS and JDK_JAVA_OPTIONS") {
+    // Adapted from https://stackoverflow.com/a/496849
+    def setEnvVars(newEnv: Map[String, String]): Unit = {
+      val classes = classOf[Collections].getDeclaredClasses
+      val env     = System.getenv()
+      for (cl <- classes)
+        if (cl.getName.equals("java.util.Collections$UnmodifiableMap")) {
+          val field = cl.getDeclaredField("m")
+          field.setAccessible(true)
+          val obj = field.get(env)
+          val map = obj.asInstanceOf[java.util.Map[String, String]]
+          map.clear()
+          map.putAll(newEnv.asJava)
+        }
+    }
+
+    val javaOptsValues       = "  -Xmx2048m   -Dhttp.proxy=4.4.4.4   -Xss8m"
+    val jdkJavaOptionsValues = " -Xmx1024m -Dscala-cli=true  -Xss4m"
```

Comment:
I suggest removing the `Xmx` option in the tests. Using it might lead to misconceptions, making one believe that scala-cli supports `Xmx` options, when it doesn't.